storage/storage_dest.go

```diff
@@ -218,9 +220,18 @@ func (s *storageImageDestination) PutBlobWithOptions(ctx context.Context, stream
 		return info, nil
 	}
```

Comment:
There is an early return above; in that case `.filename` would not be set. 

OK, that’s somewhat theoretical. But also, nothing sets `.filename` on the `TryReusingBlobWithOptions` code path. That one is 100% a blocker.